rcongui/src/components/PlayerInfo/PlayerInfo.js

```diff
@@ -0,0 +1,267 @@
+import React from "react";
+import {Avatar, Button, Grid, Popover} from "@material-ui/core";
+import Typography from "@material-ui/core/Typography";
+import {ExpandMore} from "@material-ui/icons";
+import moment from "moment";
+import MUIDataTable from "mui-datatables";
+import PropTypes from 'prop-types';
+import {withRouter} from "react-router";
+
+const NamePopOver = ({names}) => {
+    const [anchorEl, setAnchorEl] = React.useState(null);
+
+    const handleClick = (event) => {
+        setAnchorEl(event.currentTarget);
+    };
+
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+
+    const open = Boolean(anchorEl);
+    const id = open ? 'name-popover' : undefined;
+
+    return (
+        <Grid item>
+            <Button endIcon={<ExpandMore/>} onClick={handleClick}>
+                <Typography variant={"h5"}>{names[0].name}</Typography>
+            </Button>
+            <Popover
+                id={id}
+                open={open}
+                anchorEl={anchorEl}
+                onClose={handleClose}
+                anchorOrigin={{
+                    vertical: 'bottom',
+                    horizontal: 'center',
+                }}
+                transformOrigin={{
+                    vertical: 'top',
+                    horizontal: 'center',
+                }}
+            >
+                {names.map((name) => {
+                    return <Typography>{name.name}</Typography>;
+                })}
+            </Popover>
+        </Grid>
+    );
+}
+
+const Punishment = ({punishments}) => {
+    const [myRowPerPage, setRowPerPage] = React.useState(
+        window.localStorage.getItem("punishments_row_per_page") || 50
+    );
+    const saveRowsPerPage = (rowPerPage) => {
+        window.localStorage.setItem("punishments_row_per_page", rowPerPage);
+        setRowPerPage(rowPerPage);
+    };
+    const columns = [
+        {name: "action_type", label: "Type"},
+        {name: "reason", label: "Reason"},
+        {name: "by", label: "By"},
+        {
+            name: "time",
+            label: "Time",
+            options: {
+                customBodyRenderLite: (dataIndex) =>
+                    moment(punishments[dataIndex].time).format("ddd Do MMM HH:mm:ss"),
+            },
+        }
+    ];
+
+    const options = {
+        filter: false,
+        rowsPerPage: myRowPerPage,
+        selectableRows: "none",
+        rowsPerPageOptions: [10, 25, 50, 100, 250, 500, 1000],
+        onChangeRowsPerPage: saveRowsPerPage,
+    };
+
+    return (
+        <MUIDataTable
+            title={"Punishments"}
+            data={punishments}
+            columns={columns}
+            options={options}
+        />
+    );
+}
+
+class PlayerInfo extends React.Component {
+    static propTypes = {
+        match: PropTypes.object.isRequired,
+        location: PropTypes.object.isRequired,
+        history: PropTypes.object.isRequired
```

Comment:
I don't really understand those props. Isn't the only thing you need the steam id?